### PR TITLE
feat: restrict symbols in profile name

### LINKED ISSUE #2575: Show warning message if user trying to save with checkmark or other prohibited emojis/symbols
Image

### PR DESCRIPTION
**Pull request by foolo**
## What does this PR do?

- Sanitize input when selecting a profile name in settings.
- Filter out any restricted symbols when rendering existing profile names.

Notes/questions:
- Not sure about the exact list of symbols that should be restricted. I added these two at the moment: ✅🔵
- When rendering a profile name that already has a restricted symbol, what should it be replaced with? Empty string, space, something else?
- UX: What should these symbols be called in user facing texts? Restricted/prohibited/forbidden/protected, etc?

### Before

before Screenshot_2023-05-14_23-52-25

### After

after Screenshot_2023-05-14_23-52-39

### Settings:

Screenshot_2023-05-15_00-03-24

## Related issues

Fixes #2575

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Enhancement (non-breaking small changes to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### <samp>🤖 Generated by Copilot at 64be33f</samp>

*  Define constants and function for profile name validation and filtering ([link](https://github.com/lensterxyz/lenster/pull/2858/files?diff=unified&w=0#diff-6ec8aa7fb4c2ed1478249a38aabfb5f5b619d73fbe771513d05679e976037348R99-R106), [link](https://github.com/lensterxyz/lenster/pull/2858/files?diff=unified&w=0#diff-23233a7088f51f31040a7a6876916a0e8428dd087112e7affead8b92e9a87f82R1-R16))
*  Import and apply `filterProfileName` function to various components that display profile names ([link](https://github.com/lensterxyz/lenster/pull/2858/files?diff=unified&w=0#diff-2ce189b37041c30ccd1e2909ceecbb30ea9b59738f19f0311bf70e40993bca6bR13), [link](https://github.com/lensterxyz/lenster/pull/2858/files?diff=unified&w=0#diff-2ce189b37041c30ccd1e2909ceecbb30ea9b59738f19f0311bf70e40993bca6bL65-R67), [link](https://github.com/lensterxyz/lenster/pull/2858/files?diff=unified&w=0#diff-86a42fd57d0cd5557caf0ba432c3f36f048b7f3098460771c3f65de779dd14a4R6), [link](https://github.com/lensterxyz/lenster/pull/2858/files?diff=unified&w=0#diff-86a42fd57d0cd5557caf0ba432c3f36f048b7f3098460771c3f65de779dd14a4L60-R62), [link](https://github.com/lensterxyz/lenster/pull/2858/files?diff=unified&w=0#diff-b4d741b1dd3ca121615a3d6c04ffb6dc74d2413fb05a2af66b3ee4d5a030aac0R3), [link](https://github.com/lensterxyz/lenster/pull/2858/files?diff=unified&w=0#diff-b4d741b1dd3ca121615a3d6c04ffb6dc74d2413fb05a2af66b3ee4d5a030aac0L41-R44), [link](https://github.com/lensterxyz/lenster/pull/2858/files?diff=unified&w=0#diff-051c5aa8ed716f2a43344cf26ef7f981c4ad1884f3cf442ae394da8e5527aa04R23), [link](https://github.com/lensterxyz/lenster/pull/2858/files?diff=unified&w=0#diff-051c5aa8ed716f2a43344cf26ef7f981c4ad1884f3cf442ae394da8e5527aa04L121-R122), [link](https://github.com/lensterxyz/lenster/pull/2858/files?diff=unified&w=0#diff-83461f5ef7eadb62802de7bdb0ab33f1909987c38833353fcc5ccac108b678aaR3), [link](https://github.com/lensterxyz/lenster/pull/2858/files?diff=unified&w=0#diff-83461f5ef7eadb62802de7bdb0ab33f1909987c38833353fcc5ccac108b678aaL61-R63), [link](https://github.com/lensterxyz/lenster/pull/2858/files?diff=unified&w=0#diff-cc605dc6510d575486ca4cc4d51a003ca10dfeb338fced8f413d3c60d1543c90L9-R15), [link](https://github.com/lensterxyz/lenster/pull/2858/files?diff=unified&w=0#diff-120dcfc86df6670a018eec8b352b7f9a6d37aa7577510e00911662ef075577bfR5), [link](https://github.com/lensterxyz/lenster/pull/2858/files?diff=unified&w=0#diff-120dcfc86df6670a018eec8b352b7f9a6d37aa7577510e00911662ef075577bfL141-R142), [link](https://github.com/lensterxyz/lenster/pull/2858/files?diff=unified&w=0#diff-047c8c7a24c0690a4c7dc5539d56b588f13ef88bf1ce0c786f90e33b9907f3b1R14), [link](https://github.com/lensterxyz/lenster/pull/2858/files?diff=unified&w=0#diff-047c8c7a24c0690a4c7dc5539d56b588f13ef88bf1ce0c786f90e33b9907f3b1L209-R210), [link](https://github.com/lensterxyz/lenster/pull/2858/files?diff=unified&w=0#diff-e4705a2c6b2e41d8d46fac443009008c0b907f39e11267944285a4be658acf9bR3), [link](https://github.com/lensterxyz/lenster/pull/2858/files?diff=unified&w=0#diff-e4705a2c6b2e41d8d46fac443009008c0b907f39e11267944285a4be658acf9bL32-R33), [link](https://github.com/lensterxyz/lenster/pull/2858/files?diff=unified&w=0#diff-4cdf48abc8f0279e8ce53beea7dd66d71d57065e549bdf126de7bf715de38567R6), [link](https://github.com/lensterxyz/lenster/pull/2858/files?diff=unified&w=0#diff-4cdf48abc8f0279e8ce53beea7dd66d71d57065e549bdf126de7bf715de38567L64-R66), [link](https://github.com/lensterxyz/lenster/pull/2858/files?diff=unified&w=0#diff-76ec0042feeee3e7c5b48571fa583a0224d6d991cc43d81004753a2c461f38a9R5), [link](https://github.com/lensterxyz/lenster/pull/2858/files?diff=unified&w=0#diff-76ec0042feeee3e7c5b48571fa583a0224d6d991cc43d81004753a2c461f38a9L81-R82))
*  Import and use `PROFILE_NAME_VALIDATOR_REGEX` constant to validate profile name input in `editProfileSchema` object in `apps/web/src/components/Settings/Profile/Profile.tsx` ([link](https://github.com/lensterxyz/lenster/pull/2858/files?diff=unified&w=0#diff-cc605dc6510d575486ca4cc4d51a003ca10dfeb338fced8f413d3c60d1543c90L53-R65))

## Emoji

🚫🧹🛡️

### CODE REVIEW

File: packages/data/constants.ts [hunk 1]
```diff
@@ -96,6 +96,14 @@ export const URL_REGEX =
   /^(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[\da-z]+([.\-][\da-z]+)*\.[a-z]{2,63}(:\d{1,5})?(\/.*)?$/;
 export const ADDRESS_REGEX = /^(0x)?[\da-f]{40}$/i;
 export const HANDLE_REGEX = /^[\da-z]+$/;
+export const RESTRICTED_SYMBOLS = '✅🔵';
```
Comment (bigint) [inline packages/data/constants.ts:99]:
```suggestion
export const RESTRICTED_SYMBOLS = '☑️✓✔✅';
```

File: packages/lib/filterProfileName.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ * Remove restricted symbols from profile name
+ *
+ * @param name Profile name
+ * @returns Profile name with restricted symbols removed
+ */
+const filterProfileName = (name: string | null | undefined): string | null => {
+  if (!name) {
+    return null;
+  }
+  return name.replace(PROFILE_NAME_FILTER_REGEX, ' ');
```
Comment (bigint) [inline packages/lib/filterProfileName.ts:13]:
```suggestion
  }

  return name.replace(PROFILE_NAME_FILTER_REGEX, ' ');
```

Comment (bigint) [reply packages/lib/filterProfileName.ts:13]: Also add `.trim()` after replace!

Comment (foolo) [reply packages/lib/filterProfileName.ts:13]: Ahh, true! Btw, I wasn't sure what character we want to replace with, so I went with a simple space.

**Review by bigint — APPROVED** [review]
Amazing work 🚀

**Comment by bigint** [comment]
I'll write test cases in follow up PR 🙇🏼

**Pull request merged**